api/auth/auth-middleware.js

```diff
@@ -1,49 +1,84 @@
+const User = require("../users/users-model");
+const bcrypt = require("bcryptjs");
 /*
   If the user does not have a session saved in the server
-
-  status 401
-  {
-    "message": "You shall not pass!"
-  }
+  status 401 {"message": "You shall not pass!"}
 */
 function restricted() {
-
+  return async (req, res, next) => {
+    try {
+      if (!req.session && !req.session.user) {
+        return res.status(401).json({ message: "You shall not pass!" });
+      }
+      next();
+    } catch (err) {
+      next(err);
+    }
+  };
 }
 
 /*
   If the username in req.body already exists in the database
-
-  status 422
-  {
-    "message": "Username taken"
-  }
+  status 422 {"message": "Username taken"}
 */
 function checkUsernameFree() {
-
+  return async (req, res, next) => {
+    try {
+      const { username } = req.body;
+      const user = await User.findBy({ username });
```

Comment:
incearca `find` in loc de `findBy`

Reply:
am incercat, nu a mers

Reply:
am in vedere nu Users.find, ci me model .find a ii dai

Reply:
dar daca faci console log la user ce zice?

Reply:
se afiseaza array gol user[]

Reply:
> am in vedere nu Users.find, ci me model .find a ii dai

nu inteleg ce aveti in vedere? sa fac named import de {User} si sa-i aplic direct metoda find??

Reply:
pai e corect nu? el nu gaseste niciunul

Reply:
> nu inteleg ce aveti in vedere? sa fac named import de {User} si sa-i aplic direct metoda find??

nu conteaza, ai facut asa cum trebuia

Reply:
pai tu te astepti sa nu gaseasca nici un utilizator asai? si el iti da un array gol, ceea ce e corect, nu?

Reply:
da, corect. dar atunci de ce apare mesajul "User taken" ?

Reply:
![image](https://user-images.githubusercontent.com/51832550/110959823-c2967b00-8356-11eb-8fdd-97d635c9fbca.png)

Reply:
mi-am dat seama, daca user e un array, atunci in if trebuie de pus altfel conditia if(user.length>0){}

Reply:
pentru ca user[] mereu va fi true